### PR TITLE
[VDO-5738] Add perl support for running dmtest tests

### PR DESCRIPTION
**Pull request by bjohnsto**
This is the initial perl framework for running dmtest-python tests.

### CODE REVIEW

File: src/perl/dmtest/dmtests.suites [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+##
+
+# Basic VDO functional tests using filesystems
+my $CCA = "--clientClass=ALBIREO-PMI\\,RAWHIDE";
+my $CCV = "--clientClass=VDO-PMI\\,RAWHIDE";
+
+$aliasNames{Checkin} = "Grouped   --dmtestName=vdo*";
+$aliasNames{Full}    = "Separated --dmtestName=*";
+$aliasNames{Jenkins} = "Grouped   --dmtestName=vdo*";
+$aliasNames{HDD}     = "Separated --dmtestName=vdo* $CCA";
```
Comment (raeburn) [inline src/perl/dmtest/dmtests.suites:17]: Isn't dmtestName taking a regular expression to match a substring? The star at the end is both wrong (dot-star if you want to mean an arbitrary string) and unneeded ("vdo" will match anything containing "vdo"; "vdo*" matches "vd" followed by any number of o's).

File: src/perl/dmtest/DMTest/Grouped.pm [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+use base qw(DMTest);
+
+my $log = Log::Log4perl->get_logger(__PACKAGE__);
+
+#############################################################################
+# @paramList{getProperties}
+our %PROPERTIES
+  = (
+     # @ple the starting device setup on the machine
+     dmtestRepo      => "git+ssh://bjohnston\@vdo-builder-bjohnston/~/git/dmtest-python",
```
Comment (raeburn) [inline src/perl/dmtest/DMTest/Grouped.pm:35]: This says to get at the repo we should log into your account, probably not what you want :-)

Comment (bjohnsto) [reply src/perl/dmtest/DMTest/Grouped.pm:35]: oh shoot. will fix

File: src/perl/dmtest/dmtests.suites (hunk 1 above)
Comment (bjohnsto) [reply src/perl/dmtest/dmtests.suites]: updated

File: src/perl/dmtest/DMTest/Grouped.pm (hunk 2 above)
Comment (bjohnsto) [reply src/perl/dmtest/DMTest/Grouped.pm]: fixed. as well as another fix relating to the actual git clone line

**Review by raeburn — APPROVED** [review]

**Review by lorelei-sakai — COMMENTED** [review]
This would probably be easier to review if you had one commit that copied VDOTest.pm and another that did the changes, to make it clear what's new to support dmtests. But it's since this is already approved, you don't have to respell it.

**Pull request merged**